src/main/webapp/query-control.js

```diff
@@ -193,3 +201,32 @@ function createStateDropdownList() {
     datalist.appendChild(optionElem);
   });
 }
+
+// On page load, show appropriate icon based on the localStorage
+function loadAppropriateIcon() {
```

Comment:
This function name is much more generic than what it does. But I think that's a good thing, because it keeps it open for us to also use this for other things later (like logging in/out, maybe). However, in that case I think you should rename it to "loadAppropriateIcons" (plural).